File: tests/gpg/t-trustlist.c
/* t-trustlist.c - Regression test.
   Copyright (C) 2000 Werner Koch (dd9jn)
   Copyright (C) 2001, 2003, 2004 g10 Code GmbH

   This file is part of GPGME.

   GPGME is free software; you can redistribute it and/or modify it
   under the terms of the GNU Lesser General Public License as
   published by the Free Software Foundation; either version 2.1 of
   the License, or (at your option) any later version.

   GPGME is distributed in the hope that it will be useful, but
   WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
   Lesser General Public License for more details.

   You should have received a copy of the GNU Lesser General Public
   License along with this program; if not, write to the Free Software
   Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
   02111-1307, USA.  */

/* We need to include config.h so that we know whether we are building
   with large file system (LFS) support. */
#ifdef HAVE_CONFIG_H
#include <config.h>
#endif

#include <stdlib.h>
#include <stdio.h>
#include <string.h>

#include <gpgme.h>

#include "t-support.h"


int
main (int argc, char *argv[])
{
  gpgme_ctx_t ctx;
  gpgme_error_t err;
  gpgme_trust_item_t item;

  (void)argc;
  (void)argv;

  init_gpgme (GPGME_PROTOCOL_OpenPGP);

  err = gpgme_new (&ctx);
  fail_if_err (err);

  err = gpgme_op_trustlist_start (ctx, "alice", 0);
  fail_if_err (err);

  while (!(err = gpgme_op_trustlist_next (ctx, &item)))
    {
      printf ("l=%d k=%s t=%d o=%s v=%s u=%s\n",
	      item->level, item->keyid, item->type, item->owner_trust,
	      item->validity, item->name);
      gpgme_trust_item_unref (item);
    }
  if (gpgme_err_code (err) != GPG_ERR_EOF)
    fail_if_err (err);

  gpgme_release (ctx);
  return 0;
}

File: lang/python/tests/t-trustlist.py
#!/usr/bin/env python

# Copyright (C) 2016 g10 Code GmbH
#
# This file is part of GPGME.
#
# GPGME is free software; you can redistribute it and/or modify it
# under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# GPGME is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
# or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General
# Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public
# License along with this program; if not, see <http://www.gnu.org/licenses/>.

from __future__ import absolute_import, print_function, unicode_literals
del absolute_import, print_function, unicode_literals

from pyme import core, constants
import support

support.init_gpgme(constants.PROTOCOL_OpenPGP)
c = core.Context()

def dump_item(item):
    print("l={} k={} t={} o={} v={} u={}".format(
        item.level, item.keyid, item.type, item.owner_trust,
        item.validity, item.name))

c.op_trustlist_start("alice", 0)
while True:
    item = c.op_trustlist_next()
    if not item:
        break
    dump_item(item)
c.op_trustlist_end()

for item in c.op_trustlist_all("alice", 0):
    dump_item(item)
